pandas/tests/frame/test_constructors.py

```diff
@@ -1340,6 +1340,31 @@ def test_constructor_list_of_namedtuples(self):
         result = DataFrame(tuples, columns=["y", "z"])
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.skipif(not PY36, reason="Guaranteed dict order is Python>=3.7")
+    def test_constructor_list_of_dict_order(self):
+        # GH10056
+        data = [
+            {"First": 1, "Second": 4, "Third": 7, "Fourth": 10},
+            {"Second": 5, "First": 2, "Fourth": 11, "Third": 8},
+            {"Second": 6, "First": 3, "Fourth": 12, "Third": 9, "YYY": 14, "XXX": 13},
```

Comment:
What is the order here if YYY precedes Second? Somewhat wary of making assertions about ordering in 2D

Reply:
undefined.

Reply:
yeah, it's tricky... as mentioned in my general comment, we could probably use `dict.update()` as a guide. As I understand it the guiding principle is insertion order - whatever is inserted first stays where it is. You don't have to deal with reordering things that were already inserted, just tack the new keys on to the end in the order you get them.

Reply:
btw, I really appreciate how responsive you are to PRs. Just wanted to say thank you.